### PR TITLE
main page loader icon migration

### PR DESCRIPTION
**Pull request by avzz-19**

### CODE REVIEW

File: tests/integration/components/loading-spinner-test.js
Comment (future-pirate-king) [inline tests/integration/components/loading-spinner-test.js:1]: file name should be `main-loader-test.js`

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: tests/integration/components/loading-spinner-test.js
Comment (avzz-19) [reply tests/integration/components/loading-spinner-test.js:1]: done

File: app/components/main-loader/index.hbs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+>
+  {{#if (eq this.currentLoader 1)}}
+    <AkSvg::LoaderImage1 />
+  {{/if}}
+  {{#if (eq this.currentLoader 2)}}
+    <AkSvg::LoaderImage2 />
+  {{/if}}
+  {{#if (eq this.currentLoader 3)}}
+    <AkSvg::LoaderImage3 />
+  {{/if}}
```
Comment (future-pirate-king) [inline app/components/main-loader/index.hbs:17]: use `component` helper to simplify
Eg:
```
{{#let (component this.currentLoaderImage) as |LoaderImage|}}
  <LoaderImage />
{{/let}}
```

File: app/components/main-loader/index.hbs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  <AkLoader::Linear
+    @variant='determinate'
+    @progress={{this.dummyPositionIndicator}}
+    local-class='progress-container'
+    main-loader-linear-progress
+  />
+  {{!-- <AkLoader::Linear
```
Comment (future-pirate-king) [inline app/components/main-loader/index.hbs:24]: remove comment

File: app/components/main-loader/index.hbs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  {{/if}}
+  {{#if (eq this.currentLoader 2)}}
+    <AkSvg::LoaderImage2 />
+  {{/if}}
+  {{#if (eq this.currentLoader 3)}}
+    <AkSvg::LoaderImage3 />
+  {{/if}}
+  <AkLoader::Linear
+    @variant='determinate'
+    @progress={{this.dummyPositionIndicator}}
```
Comment (future-pirate-king) [inline app/components/main-loader/index.hbs:20]: rename this variable to `this.fakeProgress`

File: app/components/main-loader/index.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+export interface MainLoaderSignature {
+  Element: HTMLElement;
+}
+
+export default class MainLoaderComponent extends Component<MainLoaderSignature> {
+  @tracked dummyPositionIndicator = 0;
+  @tracked currentLoader = 1;
```
Comment (future-pirate-king) [inline app/components/main-loader/index.ts:11]: rename this to `currentLoaderImage`

File: app/components/main-loader/index.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+
+    this.updateDummyIndicatorTask();
+  }
+
+  @action
+  updateDummyIndicatorTask() {
```
Comment (future-pirate-king) [inline app/components/main-loader/index.ts:20]: rename this function to  `generateFakeProgress`

File: app/components/main-loader/index.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        this.dummyPositionIndicator += 15;
+      } else {
+        clearInterval(interval);
+      }
+    }, 600);
+
+    setInterval(() => {
+      this.switchImage();
+    }, 1200);
+  }
```
Comment (future-pirate-king) [inline app/components/main-loader/index.ts:32]: clear all intervals in `willDestroy`

File: app/components/ak-svg/loader-image1.hbs
Comment (future-pirate-king) [inline app/components/ak-svg/loader-image1.hbs:1]: rename this file to `main-loader-image1`

File: app/components/ak-svg/loader-image2.hbs
Comment (future-pirate-king) [inline app/components/ak-svg/loader-image2.hbs:1]: rename this file to `main-loader-image2`

File: app/components/ak-svg/loader-image3.hbs
Comment (future-pirate-king) [inline app/components/ak-svg/loader-image3.hbs:1]: rename this file to `main-loader-image3`

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/main-loader/index.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  switchImage() {
+    setInterval(() => {
+      this.currentLoaderImageIndex =
+        (this.currentLoaderImageIndex + 1) % this.loaderImages.length;
+    }, 1000);
+  }
+
+  willDestroy() {
+    super.willDestroy();
+  }
```
Comment (future-pirate-king) [inline app/components/main-loader/index.ts:46]: need to clear intervals in `willDestroy`
```
clearInterval(this.fakeProgressInterval)  // set this variable in `generateFakeProgress`
clearInterval(this.switchImageInterval)  // set this variable in `switchImage`
```

File: app/components/main-loader/index.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+    this.generateFakeProgress();
+  }
+
+  @action
+  generateFakeProgress() {
+    setInterval(() => {
+      if (this.fakeProgress < 100) {
+        this.fakeProgress += 15;
+      }
```
Comment (future-pirate-king) [inline app/components/main-loader/index.ts:30]: clear interval in else block

File: app/components/main-loader/index.hbs [hunk 9]
```diff
@@ -0,0 +1,18 @@
+<AkStack
+  @justifyContent='center'
+  @alignItems='center'
+  @direction='column'
+  @spacing='1'
+  local-class='loading-container'
+  main-loader-container
```
Comment (future-pirate-king) [inline app/components/main-loader/index.hbs:7]: rename to `data-test-main-loader-container`

File: app/components/main-loader/index.hbs [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+>
+  {{#let (component this.currentLoaderImage) as |LoaderImage|}}
+    <LoaderImage />
+  {{/let}}
+  <AkLoader::Linear
+    @variant='determinate'
+    @progress={{this.fakeProgress}}
+    local-class='progress-container'
+    main-loader-linear-progress
```
Comment (future-pirate-king) [inline app/components/main-loader/index.hbs:16]: rename to `data-test-main-loader-linear-progress`

File: app/components/main-loader/index.hbs [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+  @justifyContent='center'
+  @alignItems='center'
+  @direction='column'
+  @spacing='1'
+  local-class='loading-container'
+  main-loader-container
+>
+  {{#let (component this.currentLoaderImage) as |LoaderImage|}}
+    <LoaderImage />
+  {{/let}}
```
Comment (future-pirate-king) [inline app/components/main-loader/index.hbs:11]: add a new line

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: tests/integration/components/ak-loader/linear-test.js [hunk 12]
```diff
@@ -43,16 +43,18 @@ module('Integration | Component | ak-loader/linear', function (hooks) {
       '[data-test-ak-loader-linear-indicator]'
     );
 
-    assert
-      .dom(linearProgressIndicator)
-      .exists()
-      .hasStyle({ left: `${this.progress - this.width}px` });
+    assert.dom(linearProgressIndicator).exists();
```
Comment (future-pirate-king) [inline tests/integration/components/ak-loader/linear-test.js:46]: add new line

File: app/components/main-loader/index.ts [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+  @action
+  switchImage() {
+    this.switchImageInterval = setInterval(() => {
+      this.currentLoaderImageIndex =
+        (this.currentLoaderImageIndex + 1) % this.loaderImages.length;
+    }, 1000);
+  }
+
+  willDestroy() {
+    super.willDestroy();
```
Comment (future-pirate-king) [inline app/components/main-loader/index.ts:58]: add new line

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

**Review by future-pirate-king — APPROVED** [review]

**Pull request merged**